cobc/pplex.l

```diff
@@ -1936,42 +1936,38 @@ is_cobol_word_char (const char c)
 	return c == '-' || c == '_' || isalnum (c);
 }
 
-static char *
-get_word (const char *str, char *word_buff)
+static int
+get_word (const char *str, char *word_buff, size_t buff_len)
 {
-	char *ret = NULL;
+	int length = 0;
 	if (str) {
-		int length = COB_MAX_WORDLEN;
 		while (*str == ' ' || *str == ',' || *str == ';') {
 			++str;
 		}
-		ret = word_buff;
-		while (is_cobol_word_char(*str)
-		    && length--) {
+		while (length < buff_len && is_cobol_word_char(*str)){
 			*word_buff++ = (unsigned char)toupper((unsigned char)*str++);
+			length++;
 		}
-		if (length != 0) {
+		if (length < buff_len) {
 			*word_buff = 0;
 		} else {
-			ret = NULL;
+			length = 0;
 		}
 	}
-	return ret;
+	return length;
 }
 
 static int
 is_condition_directive_clause (const char *buff)
 {
-	char word_buff[COB_MAX_WORDLEN + 1];
-	char *p;
+	char p[COB_MAX_WORDLEN + 1];
```

Comment:
while keeping `word_buff` (actually `buff` is enough) it only needs to be of size 14 (`END-EVALUATE` is the longest one) and down in the call `sizeof(buff)` can be used then

[I guess this good commit was mostly for checking the CI result, so the compare was not changed to `len == 3 && memcmp (buff, "IF", 3)`]

Reply:
New commit adds this.